core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -239,10 +269,36 @@ func (l *logPollerWrapper) LatestEvents() ([]evmRelayTypes.OracleRequest, []evmR
 		}
 	}
 
-	l.lggr.Debugw("LatestEvents: done", "nRequestLogs", len(resultsReq), "nResponseLogs", len(resultsResp), "nextBlock", nextBlock, "latest", latest)
+	l.lggr.Debugw("LatestEvents: done", "nRequestLogs", len(resultsReq), "nResponseLogs", len(resultsResp), "startBlock", startBlock, "endBlock", latest)
 	return resultsReq, resultsResp, nil
 }
 
+func (l *logPollerWrapper) FilterPreviouslyDetectedEvents(logs []logpoller.Log, detectedEvents *detectedEvents) []logpoller.Log {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	filteredLogs := []logpoller.Log{}
+	for _, log := range logs {
+		var requestId [32]byte
+		copy(requestId[:], log.Topics[1]) // requestId is the second topic (1st topic is the event signature)
+		if _, ok := detectedEvents.isPreviouslyDetected[requestId]; !ok {
+			filteredLogs = append(filteredLogs, log)
+			detectedEvents.isPreviouslyDetected[requestId] = true
+			detectedEvents.detectedEventsOrdered = append(detectedEvents.detectedEventsOrdered, detectedEvent{requestId: requestId, timeDetected: time.Now()})
+		}
+	}
+	expiredRequests := 0
+	for _, detectedEvent := range detectedEvents.detectedEventsOrdered {
+		if detectedEvent.timeDetected.Unix() < time.Now().Unix()-l.logPollerCacheDurationSec {
```

Comment:
I think sorting them by time is fine but would it be cleaner if we sorted by block number instead? Not requesting any changes, just asking to think about it if you haven't.

Reply:
I did think about this, but chose otherwise as I didn't want to have to reason through the ramifications of the latest block decreasing during some complex reorg.

Reply:
Keep in mind that the sorted list doesn't have to perfectly sorted all the time ;) Worst case we'll briefly accumulate a few more entries and then remove a larger chunk at once.

Reply:
Understood, however I think it is fine to leave this as a timestamp for now unless we see a good reason to alter this again.